### PR TITLE
Contracts grammar check dev1.5.0

### PR DESCRIPTION
**Pull request by comeonbuddy**
### Please check if the PR fulfills these requirements
- [x] The commit message follows our Submission guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? 
grammar fixes

### What is the current behavior? 
we have some grammar issues 

### Does this PR introduce a breaking change?
(What changes might users need to make in their application due to this PR?)

### CODE REVIEW

File: contracts/interfaces/IModuleFactory.sol [hunk 1]
```diff
@@ -57,27 +57,27 @@ interface IModuleFactory {
     function getTags() external view returns (bytes32[]);
 
     /**
-     * @notice used to change the fee of the setup cost
-     * @param _newSetupCost new setup cost
+     * @notice Used to change the setup fee
+     * @param _newSetupCost New setup cost
```
Comment (maxsam4) [inline contracts/interfaces/IModuleFactory.sol]: I think we should call it setup cost or setup fee in both lines for consistency. Same with a few functions below.

File: contracts/mocks/PolyTokenFaucet.sol [hunk 2]
```diff
@@ -71,7 +71,7 @@ contract PolyTokenFaucet {
     }
 
     /**
-     * @notice `balanceOf` function to get the balance of token holders
+     * @notice Returns the balance of a token holders
```
Comment (maxsam4) [inline contracts/mocks/PolyTokenFaucet.sol]:
```suggestion
     * @notice Returns the balance of a token holder
```

File: contracts/mocks/PolyTokenFaucet.sol [hunk 3]
```diff
@@ -80,7 +80,7 @@ contract PolyTokenFaucet {
     }
 
     /**
-     * @notice `msg.sender` approves `_spender` to spend `_value` tokens
+     * @notice Used for `msg.sender` to approve `_spender` to spend `_value` tokens
```
Comment (maxsam4) [inline contracts/mocks/PolyTokenFaucet.sol]:
```suggestion
     * @notice Used by `msg.sender` to approve `_spender` to spend `_value` tokens
```
I think this makes more sense

File: contracts/modules/Burn/TrackedRedemptionFactory.sol [hunk 4]
```diff
@@ -27,12 +27,12 @@ contract TrackedRedemptionFactory is ModuleFactory {
...
     /**
-     * @notice used to launch the Module with the help of factory
-     * @return address Contract address of the Module
+     * @notice Used to launch the Module with the help of factory
+     * @return Address Contract address of the Module
      */
     function deploy(bytes /* _data */) external returns(address) {
         if (setupCost > 0)
-            require(polyToken.transferFrom(msg.sender, owner, setupCost), "Failed transferFrom because of sufficent Allowance is not provided");
+            require(polyToken.transferFrom(msg.sender, owner, setupCost), "Failed transferFrom due to insufficent Allowance provided");
```
Comment (maxsam4) [inline contracts/modules/Burn/TrackedRedemptionFactory.sol]:
```suggestion
            require(polyToken.transferFrom(msg.sender, owner, setupCost), "Insufficent allowance or balance");
```
Shorter and more technical correctness.

File: contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol [hunk 5]
```diff
@@ -114,10 +114,10 @@ contract ERC20DividendCheckpoint is DividendCheckpoint {
         internal  
     {
         require(_excluded.length <= EXCLUDED_ADDRESS_LIMIT, "Too many addresses excluded");
-        require(_expiry > _maturity, "Expiry is before maturity");
-        require(_expiry > now, "Expiry is in the past");
+        require(_expiry > _maturity, "Invalid expiry or/and maturity");
```
Comment (maxsam4) [inline contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol]: I think the earlier one was easier to understand.

File: contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol [hunk 6]
```diff
@@ -114,10 +114,10 @@ contract ERC20DividendCheckpoint is DividendCheckpoint {
         internal  
     {
         require(_excluded.length <= EXCLUDED_ADDRESS_LIMIT, "Too many addresses excluded");
-        require(_expiry > _maturity, "Expiry is before maturity");
-        require(_expiry > now, "Expiry is in the past");
+        require(_expiry > _maturity, "Invalid expiry or/and maturity");
+        require(_expiry > now, "Invalid expiry");
         require(_amount > 0, "No dividend sent");
-        require(_token != address(0), "0x not valid token");
+        require(_token != address(0), "0x is not a valid token");
```
Comment (maxsam4) [inline contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol]:
```suggestion
        require(_token != address(0), "Invalid token");
```

We need to keep the strings as short as possible. Dimitriy's branch is running out of gas because of reason strings.

File: contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol [hunk 7]
```diff
@@ -181,8 +181,8 @@ contract ERC20DividendCheckpoint is DividendCheckpoint {
      */
     function reclaimDividend(uint256 _dividendIndex) external withPerm(MANAGE) {
         require(_dividendIndex < dividends.length, "Incorrect dividend index");
-        require(now >= dividends[_dividendIndex].expiry, "Dividend expiry is in the future");
-        require(!dividends[_dividendIndex].reclaimed, "Dividend already claimed");
+        require(now >= dividends[_dividendIndex].expiry, "Invalid dividend expiry.");
+        require(!dividends[_dividendIndex].reclaimed, "Dividend has already been claimed");
```
Comment (maxsam4) [inline contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol]: the previous one is ok imo. It's shorter and hence, saves gas. This contract and Security token contract are really on the border of gas limit. Need to keep strings as short as possible on both of these.

File: contracts/modules/Checkpoint/EtherDividendCheckpointFactory.sol [hunk 8]
```diff
@@ -27,12 +27,12 @@ contract EtherDividendCheckpointFactory is ModuleFactory {
...
 
     /**
-     * @notice used to launch the Module with the help of factory
+     * @notice Used to launch the Module with the help of factory
      * @return address Contract address of the Module
      */
     function deploy(bytes /* _data */) external returns(address) {
         if(setupCost > 0)
-            require(polyToken.transferFrom(msg.sender, owner, setupCost), "Failed transferFrom because of sufficent Allowance is not provided");
+            require(polyToken.transferFrom(msg.sender, owner, setupCost), "Failed transferFrom due to insufficent Allowance provided");
```
Comment (maxsam4) [inline contracts/modules/Checkpoint/EtherDividendCheckpointFactory.sol]:
```suggestion
            require(polyToken.transferFrom(msg.sender, owner, setupCost), "Insufficent allowance or balance");
```

Please change the remaining `transferFrom` messages as well

File: contracts/modules/STO/PreSaleSTO.sol [hunk 9]
```diff
@@ -74,8 +74,8 @@ contract PreSaleSTO is ISTO {
      */
     function allocateTokens(address _investor, uint256 _amount, uint256 _etherContributed, uint256 _polyContributed) public withPerm(PRE_SALE_ADMIN)
     {
-        require(now <= endTime, "Current time should less than the endTime");
-        require(_amount > 0, "No. of tokens provided should be greater the zero");
+        require(now <= endTime, "EndTime is in the past");
+        require(_amount > 0, "No. of tokens provided should be greater than zero");
```
Comment (maxsam4) [inline contracts/modules/STO/PreSaleSTO.sol:78]: For consistency, change it to what went wrong

File: contracts/modules/STO/USDTieredSTO.sol [hunk 10]
```diff
@@ -217,10 +217,10 @@ contract USDTieredSTO is ISTO, ReentrancyGuard {
...
         require(_tokensPerTierDiscountPoly.length == _tokensPerTierTotal.length, "Mismatch b/w discount tokens / tier & tokens / tier");
         for (uint8 i = 0; i < _ratePerTier.length; i++) {
-            require(_ratePerTier[i] > 0, "Rate > 0");
-            require(_tokensPerTierTotal[i] > 0, "Tokens per tier > 0");
-            require(_tokensPerTierDiscountPoly[i] <= _tokensPerTierTotal[i], "Discounted tokens / tier <= to tokens / tier");
-            require(_ratePerTierDiscountPoly[i] <= _ratePerTier[i], "Discounted rate / tier <= rate / tier");
+            require(_ratePerTier[i] > 0, "Rate is not larger than 0");
+            require(_tokensPerTierTotal[i] > 0, "Tokens per tier is not larger than 0");
+            require(_tokensPerTierDiscountPoly[i] <= _tokensPerTierTotal[i], "Discounted tokens / tier is larger than tokens / tier");
+            require(_ratePerTierDiscountPoly[i] <= _ratePerTier[i], "Discounted rate / tier is larger than rate / tier");
```
Comment (maxsam4) [inline contracts/modules/STO/USDTieredSTO.sol]: This contract is on verge of going over the limit as well. We might have to revert these changes if it does end up going over the limit. changes are good otherwise.

**Review by maxsam4 — CHANGES_REQUESTED** [review]

File: contracts/interfaces/IModuleFactory.sol [hunk 11]
```diff
@@ -57,27 +57,27 @@ interface IModuleFactory {
...
+     * @notice Used to change the subscription fee
+     * @param _newSubscriptionCost New subscription cost
      */
     function changeFactorySubscriptionFee(uint256 _newSubscriptionCost) external;
 
     /**
      * @notice Function use to change the lower and upper bound of the compatible version st
      * @param _boundType Type of bound
-     * @param _newVersion new version array
+     * @param _newVersion New versions array
```
Comment (satyamakgec) [inline contracts/interfaces/IModuleFactory.sol]: It will be `New version`

File: contracts/modules/STO/USDTieredSTO.sol (hunk 10 above)
Comment (satyamakgec) [reply contracts/modules/STO/USDTieredSTO.sol]: Yes keep those as it is. Because of bytecode limitations.

File: contracts/modules/STO/USDTieredSTO.sol [hunk 12]
```diff
@@ -399,7 +399,7 @@ contract USDTieredSTO is ISTO, ReentrancyGuard {
         uint256 originalUSD = investedUSD;
 
         // Check for minimum investment
-        require(investedUSD.add(investorInvestedUSD[_beneficiary]) >= minimumInvestmentUSD, "Total investment < minimumInvestmentUSD");
+        require(investedUSD.add(investorInvestedUSD[_beneficiary]) >= minimumInvestmentUSD, "Total investment is less than minimumInvestmentUSD");
```
Comment (satyamakgec) [inline contracts/modules/STO/USDTieredSTO.sol]: same here bytes code constraints

**Review by satyamakgec — CHANGES_REQUESTED** [review]
Whenever you changing the reason string please be make sure to make them concise, clean and meaningful.

File: contracts/helpers/PolyToken.sol [hunk 13]
```diff
@@ -175,14 +175,14 @@ contract PolyToken is IERC20 {
     }
 
     /**
-    * @dev Decrease the amount of tokens that an owner allowed to a spender.
+    * @dev Decreases the amount of tokens that an owner has allowed aspender to spend
```
Comment (maxsam4) [inline contracts/helpers/PolyToken.sol]:
```suggestion
    * @dev Decreases the amount of tokens that an owner has allowed a spender to spend
```

**Review by satyamakgec — APPROVED** [review]

**Review by maxsam4 — APPROVED** [review]
